c7n/resources/ecs.py

```diff
@@ -780,3 +780,119 @@ def process(self, resources, event=None):
 Service.action_registry.register('auto-tag-user', AutoTagUser)
 Task.action_registry.register('auto-tag-user', AutoTagUser)
 ContainerInstance.action_registry.register('auto-tag-user', AutoTagUser)
+
+
+@ECSCluster.action_registry.register('deregister-instances')
+class DeregisterInstances(BaseAction):
+    """Deregister container instances from an ECS Cluster.
+
+    Setting force = True will deregister the instance irrespective of task
+    running on it. Instances will still be running.
+
+    :example:
+
+      .. code-block:: yaml
+
+            policies:
+              - name: deregister-instances
+                resource: ecs
+                filters:
+                    - 'tag:Name': 'c7n'
+                actions:
+                    - type: deregister-instances
+                      DeleteCapacityProvider: True
```

Comment:
we generally only do InitCaps on pass through api parameters due to shape validation, not on configuration options. perhaps `include-provider: true`

Reply:
Thanks, switched it to `include_provider`!